src/python/pants/option/subsystem.py

```diff
@@ -58,6 +71,7 @@ def options_scope(self):
 _S = TypeVar("_S", bound="Subsystem")
 
 
+@dataclass(frozen=True)
 class Subsystem(Optionable):
```

Comment:
Does it make sense to still subclass `Optionable`? And specifically for `Optionable` to still have an `__init__.py`.

I think that generally we wanted to delete `Optionable` in favor of `Subsystem`. Benjy made several changes in that direction, but didn't finish it.

Reply:
I made a few more tweaks in here (top commit) to get tests passing. While I would be fine with removing `Optionable` at this point, #10062 is already a fairly large patch series (have one more to post), so I'd rather not do that here.

Reply:
Thanks! Totally valid to wait to remove `Optionable`.

Reply:
I would like to get rid of the distinction between `Optionable` and `Subsystem`, in favor of just `Subsystem`, but it turns out to be non-trivial. I'd say punt on that for now.